### PR TITLE
fs: unify fs.base.FileSystem with FSSpecWrapper

### PR DESCRIPTION
**Pull request by skshetry**
Since ll of our filesystems are fsspec-compliant, there's no need for compat wrapper.

* [x] ❗ I have followed the [Contributing to DVC](https://dvc.org/doc/user-guide/contributing/core) checklist.

* [x] 📖 If this PR requires [documentation](https://dvc.org/doc) updates, I have created a separate PR (or issue, at least) in [dvc.org](https://github.com/iterative/dvc.org) and linked it here.

Thank you for the contribution - we'll try to review it as soon as possible. 🙏

### CODE REVIEW

File: tests/unit/remote/test_base.py [hunk 1]
```diff
@@ -20,6 +20,7 @@ def __eq__(self, other):
 @mock.patch.object(ObjectDB, "list_hashes_exists")
 def test_hashes_exist(object_exists, traverse, dvc):
     odb = ObjectDB(FileSystem(), None)
+    odb.fs.TRAVERSE_PREFIX_LEN = 3
```
Comment (skshetry) [inline tests/unit/remote/test_base.py:23]: Still need to figure out this test.

File: tests/unit/remote/test_base.py [hunk 2]
```diff
@@ -69,6 +70,7 @@ def test_hashes_exist(object_exists, traverse, dvc):
 @mock.patch.object(ObjectDB, "_path_to_hash", side_effect=lambda x: x)
 def test_list_hashes_traverse(_path_to_hash, list_hashes, dvc):
     odb = ObjectDB(FileSystem(), None)
+    odb.fs.TRAVERSE_PREFIX_LEN = 3
```
Comment (skshetry) [inline tests/unit/remote/test_base.py:73]: Need to figure out why this is failing for TRAVERSE_PREFIX_LEN=2.

**Review by efiop — APPROVED** [review]

File: tests/unit/remote/test_base.py (hunk 2 above)
Comment (efiop) [reply tests/unit/remote/test_base.py:73]: For the record: we'll likely get rid of it in favor of `fs.find("12/3*")` or something.

File: tests/unit/remote/test_base.py [hunk 3]
```diff
@@ -34,16 +34,16 @@ def test_hashes_exist(object_exists, traverse, dvc):
     # large remote, small local
     object_exists.reset_mock()
     traverse.reset_mock()
-    with mock.patch.object(odb, "_list_hashes", return_value=list(range(256))):
+    with mock.patch.object(odb, "_list_hashes", return_value=list(range(512))):
         hashes = list(range(1000))
```
Comment (skshetry) [inline tests/unit/remote/test_base.py:38]:
```suggestion
        hashes = list(range(10000))
```

File: tests/unit/remote/test_base.py [hunk 4]
```diff
@@ -34,16 +34,16 @@ def test_hashes_exist(object_exists, traverse, dvc):
...
-        # before returning all 256 remote hashes
+        # before returning all 512 remote hashes
         max_hashes = math.ceil(
             odb._max_estimation_size(hashes)
             / pow(16, odb.fs.TRAVERSE_PREFIX_LEN)
         )
-        assert max_hashes < 256
+        assert max_hashes < 512
         object_exists.assert_called_with(
             frozenset(range(max_hashes, 1000)), None, None
```
Comment (skshetry) [inline tests/unit/remote/test_base.py:48]:
```suggestion
            frozenset(range(max_hashes, 10000)), None, None
```

**Comment by skshetry** [comment]
@pmrowla, can you please verify the changes in the tests are correct? This PR changes the default `fs.TRAVERSE_PREFIX_LEN` from 3 to 2.

**Review by pmrowla — APPROVED** [review]

**Pull request merged**